components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -0,0 +1,291 @@
+     */
+
+    public List<SecretResponse> getSecretsList(String secretType) {
+
+        try {
+            Secrets secrets = SecretManagementServiceHolder.getSecretConfigManager().getSecrets(secretType);
+            List<Secret> secretsList = secrets.getSecrets();
+            return secretsList.stream().map(secret ->
+                    buildSecretResponseFromResponseDTO(secret)).collect(Collectors.toList());
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_GETTING_SECRET, null);
+        }
+    }
+
+    /**
+     * Update secret details by name.
+     *
+     * @param name                secret name.
+     * @param secretUpdateRequest secret's updated details.
+     * @return Updated secret.
+     */
+    public SecretResponse updateSecret(String secretType, String name, SecretUpdateRequest secretUpdateRequest) {
+
+        Secret requestDTO, responseDTO;
+        SecretAddRequest secretAddRequest = buildSecretAddFromSecretUpdateRequest(name, secretUpdateRequest);
+        try {
+            requestDTO = buildSecretRequestDTOFromSecretAddRequest(secretAddRequest);
+            responseDTO = SecretManagementServiceHolder.getSecretConfigManager().replaceSecret(secretType, requestDTO);
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_SECRET,
+                    name);
+        }
+        return buildSecretResponseFromResponseDTO(responseDTO);
+    }
+
+    /**
+     * Build secretAdd object from secret update request.
+     *
+     * @param name                secret name.
+     * @param secretUpdateRequest secret's update request body.
+     * @return secretAdd object
+     */
+    private SecretAddRequest buildSecretAddFromSecretUpdateRequest(String name,
+                                                                   SecretUpdateRequest secretUpdateRequest) {
+
+        SecretAddRequest secretAddRequest = new SecretAddRequest();
+        secretAddRequest.setName(name);
+        secretAddRequest.setValue(secretUpdateRequest.getValue());
+        if (!StringUtils.isEmpty(secretUpdateRequest.getDescription())) {
+            secretAddRequest.setDescription(secretUpdateRequest.getDescription());
+        }
+        return secretAddRequest;
+    }
+
+    private APIError handleSecretMgtException(SecretManagementException e, SecretManagementConstants.ErrorMessage
+            errorEnum, String data) {
+
+        ErrorResponse errorResponse = getErrorBuilder(errorEnum, data).build(log, e, errorEnum.getDescription());
+        Response.Status status;
+        if (e instanceof SecretManagementClientException) {
+            if (e.getErrorCode() != null) {
+                String errorCode = e.getErrorCode();
+                errorCode = errorCode.contains(CONFIG_MGT_ERROR_CODE_DELIMITER) ? errorCode :
```

Comment:
Please elaborate what's the logic here? I can see that https://github.com/wso2/identity-api-server/pull/295/files#diff-14a2454eeeb347038bc033b0cd971ae0c929ffde8a3d5210731850ac33ac579eR73 appends the ERROR_PREFIX to the error code. Will this result in adding the prefix twice ?

Reply:
Already resolved with https://github.com/wso2/identity-api-server/pull/303/files